e2e/tcproute_test.go

```diff
@@ -114,18 +113,23 @@ func TestTcpRoute(t *testing.T) {
 }
 
 func buildNEG(graphBuilder *rgraph.Builder, name, zone string) (*cloud.ResourceID, error) {
-	negID := networkendpointgroup.ID(testFlags.project, meta.ZonalKey(name, zone))
+	negID := networkendpointgroup.ID(testFlags.project, meta.ZonalKey(resourceName(name), zone))
 	negMut := networkendpointgroup.NewMutableNetworkEndpointGroup(testFlags.project, negID.Key)
 	negMut.Access(func(x *compute.NetworkEndpointGroup) {
 		x.Zone = zone
+		x.NetworkEndpointType = "GCE_VM_IP_PORT"
+		x.Name = negID.Key.Name
+		x.Network = fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/global/networks/default", testFlags.project)
```

Comment:
suggest putting in e2e/testutils/testutils.go

```
type Res struct {
  Project, Region, Zone, Resource string
}

// NetworkURL returns a URL to be used for testing.
// If Project is empty, defaults to "test-project".
// If Name is empty, defaults to "default".
func NetworkURL(r *R) string {
  return fmt.Sprintf(...)
}
```

then usage:

```
  x.Network = testutils.NetworkURL(testutils.Res{Project: "xxx"}
```

this way we can 1) make it easy to write tests, 2) reuse

Reply:
I think `cloud.ResourceID` already covers it, so I added `defaultNetwork` and `defaultSubnetwork`.

Reply:
I just realized that there is `cloud.NewSubnetworkResourceID`, so it is even simpler (now).